Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java

```diff
@@ -117,6 +122,8 @@ protected void onCreate(Bundle savedInstanceState) {
         AppState.getInstance().preferences = getPreferences(MODE_PRIVATE);
         AppPreferences.applyAll();
 
+        //Ensure permissions permissions before any disk IO
+        ensurePermissions();

```

Comment:
should we do the same before creating the report?

Reply:
Nope, as this is in main activity, it truly need to be requested once.

Reply:
We could actually ask this permission only when writing the report as if the permission wasn't granted, there are no reports to remove

Reply:
is it granted once and forever?
if it's possible to grant for single app execution, then this place seems better

Reply:
it's granted until user goes to system settings and revokes the permission. But again, that thing also checks if the permission was already granted, so isn't asking again.